Add terrain_map_publisher to spirit_utils

jcnorby opened the pull request:
And implement in planning examples.

nikolaif399 commented on spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:
```diff
@@ -0,0 +1,79 @@
+#include "grid_map_core/TypeDefs.hpp"
+
+#include <string>
+
+class ImageToGridmapDemo
```
Class doxy brief required

nikolaif399 commented on spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:
```diff
@@ -0,0 +1,79 @@
+  */
+  bool readParameters();
+
+  void imageCallback(const sensor_msgs::Image& msg);
+
+  grid_map::GridMap dataSparseMap_;
```
Doxygen

nikolaif399 commented on spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:
```diff
@@ -0,0 +1,79 @@
+  //! Resolution of the grid map.
+  double resolution_;
+
+  //! Range of the height values.
+  double minHeight_;
+  double maxHeight_;
```
Separate into two doxy comments (one per var)

nikolaif399 commented on spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:
```diff
@@ -0,0 +1,79 @@
+  //! Frame id of the grid map.
+  std::string mapFrameId_;
+
+  bool mapInitialized_;
```
Doxy

nikolaif399 commented:
Need some doxygen comments on the ImageToGridMap class, otherwise looks good. I see you fixed the build warning about depending on grid_map directly.

jcnorby commented:
Oh, whoops, good catch - ImageToGridMap was supposed to be deleted, I merged it into the terrain_map_publisher class. Fixed, and updated the yaml and config files as well.

jcnorby reviewed (commented):
Can you take a look at this and approve if good?
